pkg/kopia/utils.go

```diff
@@ -184,3 +202,95 @@ func GetDataStoreGeneralMetadataCacheSize(opt map[string]int) int {
 	}
 	return defaultDataStoreGeneralMetadataCacheSizeMB
 }
+
+// SetLabelsToPodOptionsIfRequired sets labels to PodOptions
+func SetLabelsToPodOptionsIfRequired(options *kube.PodOptions) {
+	updateNeeded, labels := getKanisterPodLabels()
+	if updateNeeded {
+		if options.Labels == nil {
+			options.Labels = make(map[string]string)
+		}
+		for k, v := range *labels {
+			options.Labels[k] = v
+		}
+	}
+}
+
+func getKanisterPodLabels() (bool, *map[string]string) {
+	return parseToLabelSelector(KanisterPodCustomLabelsEnv)
+}
+
+func parseToLabelSelector(envKey string) (bool, *map[string]string) {
+	val, ok := os.LookupEnv(envKey)
+	if !ok || val == "" {
+		return false, nil
+	}
+	ls, err := metav1.ParseToLabelSelector(val)
+	if err != nil {
+		return false, nil
+	}
+	return true, &ls.MatchLabels
+}
+
+// SetAnnotationsToPodOptionsIfRequired sets annotations to PodOptions
+func SetAnnotationsToPodOptionsIfRequired(options *kube.PodOptions) {
+	updateNeeded, annotations := getKanisterPodAnnotations()
+	if updateNeeded {
+		if options.Annotations == nil {
+			options.Annotations = make(map[string]string)
+		}
+		for k, v := range *annotations {
+			options.Annotations[k] = v
+		}
+	}
+}
+
+func getKanisterPodAnnotations() (bool, *map[string]string) {
+	return parseToLabelSelector(KanisterPodCustomAnnotationsEnv)
+}
+
+// SetResourceRequirementsToPodOptionsIfRequired sets resource requirements to PodOptions
+func SetResourceRequirementsToPodOptionsIfRequired(options *kube.PodOptions) {
+	updateNeeded, res := GetResourceRequirementsForKanisterPods()
+	if updateNeeded {
+		options.Resources = *res
+	}
+}
+
+// GetResourceRequirementsForKanisterPods returns resource requirements if set in configmap
+func GetResourceRequirementsForKanisterPods() (bool, *corev1.ResourceRequirements) {
```

Comment:
```suggestion
func ResourceRequirementsForKanisterPods() (bool, *corev1.ResourceRequirements) {
```

Reply:
I think we should remove get and set from other places as well, if other's  don't have an objection.

Reply:
That make sense, but I am thinking to raise another PR to refactor these functions